src/server/pps/server/api_server.go

```diff
@@ -1422,6 +1408,48 @@ func job(jobInfo *persist.JobInfo) *batch.Job {
 	}
 }
 
+func (a *apiServer) deletePipeline(ctx context.Context, pipeline *ppsclient.Pipeline) error {
+	persistClient, err := a.getPersistClient()
+	if err != nil {
+		return err
+	}
+
+	if pipeline == nil {
+		return fmt.Errorf("Pipeline cannot be nil")
+	}
+
+	// Delete kubernetes jobs.  Otherwise we won't be able to create jobs with
+	// the same IDs, since kubernetes jobs simply use these IDs as their names
+	jobInfos, err := persistClient.ListJobInfos(ctx, &ppsclient.ListJobRequest{
+		Pipeline: pipeline,
+	})
+	if err != nil {
+		return err
+	}
+	for _, jobInfo := range jobInfos.JobInfo {
+		if err = a.kubeClient.Extensions().Jobs(a.namespace).Delete(jobInfo.JobID, nil); err != nil {
+			return err
```

Comment:
I think it makes more sense to log the error instead of exiting here.  Kubernetes jobs are in a way outside of our control; if the user manually deleted a job, they won't be able to use DeleteAll anymore.  Or we can just ignore `not found` error or something.